Restrict darcs (#1680)
dedis/cothority: byzcoin/Contracts.md

send a spawn request to an instance of a SecureDarc contract that includes the
"spawn:eventlog" rule, and the instruction must be signed by one of the keys
mentioned in the rule's expression.

The new instance spawned will have an instance ID equal to the hash of
the Spawn instruction. The client can remember this instance ID in order
to invoke methods on it later.

### Invoke

The first method that a client can invoke on a SecureDarc instance is `evolve`,
which asks ByzCoin to store a new version of the Darc. The rules may be
modified but new rules cannot be added. Use `evolve_unrestricted` to add rules.

### Delete

When a Darc instance receives a `Delete` instruction, it will be removed from
the global state.

### Secure Darc Customization

Sometimes the user needs fine-grained control of all aspects of their access
control policy. For these purposes we encourage the user to write their own
Darc contracts.

Suppose we are in a hierarchical situation where the boss is allowed to do
anything, the managers are allowed to do a certain set of tasks including
spawning user Darcs and the users are allowed to do another set of tasks but

> they are not allowed to do anything. The SecureDarc contract already achieves

Reviewer: ```suggestion
they are not allowed to spawn any new Darcs. The SecureDarc contract already achieves
```
